daft/dataframe/dataframe.py

```diff
@@ -728,7 +728,13 @@ def write_iceberg(self, table: "pyiceberg.table.Table", mode: str = "append") ->
                 partitioning[field].append(getattr(data_file.partition, field, None))
 
         if parse(pyiceberg.__version__) >= parse("0.7.0"):
-            from pyiceberg.table import ALWAYS_TRUE, PropertyUtil, TableProperties
+            from pyiceberg.table import ALWAYS_TRUE, TableProperties
+            if parse(pyiceberg.__version__) >= parse("0.8.0"):
+                from pyiceberg.utils.properties import property_as_bool
+                property_as_bool = property_as_bool
+            if parse(pyiceberg.__version__) < parse("0.8.0"):
+                from pyiceberg.table import PropertyUtil
+                property_as_bool = PropertyUtil.property_as_bool
```

Comment:
nit: could we make this instead:

```python
if parse(pyiceberg.__version__) >= parse("0.8.0"):
    from pyiceberg.utils.properties import property_as_bool
    property_as_bool = property_as_bool
else:
    from pyiceberg.table import PropertyUtil
    property_as_bool = PropertyUtil.property_as_bool
```

Reply:
@jaychia  from pyiceberg.table import PropertyUtil is add in 0.7.0 , we also should check >=0.7.0 version

Reply:
Understood -- let's throw an error then in an else clause? Would be cleaner to avoid a potential situation in which property_as_bool is not defined.